### ISSUE TITLE
BulkInsert does not provide information on failed records

### ISSUE THREAD

**Issue by frliran** [borisdj/EFCore.BulkExtensions#422]
When using BulkInsert for over 100,000 records and one of them failed, there is no output telling which record failed.
I've tried using DbUpdateException like when using EF SaveChanges() but it does not throw that exception.

**Comment by borisdj** [comment]
Could you make an example in the existing with this failure and what you would expect to be returned.

**Comment by frliran** [comment]
It is more of a concept issue rather than a specific exception.
What I meant was that while testing my code that uses your Nuget package, I tried adding some invalid records to the list that is inserted in bulk, like missing FK or invalid data, etc.
The "BulkInsert" method just throws an exception, it doesn't say which record was rejected and why, and it also doesn't give me the option to decide whether I want all 100K records to fail because of that single invalid record or not.
So I just did it on my own by splitting the list in half recursively to isolate the invalid records and I also return them as output.
